src/mlpack/tests/image_load_test.cpp

```diff
@@ -63,10 +66,114 @@ BOOST_AUTO_TEST_CASE(SaveImageAPITest)
 
   BOOST_REQUIRE_EQUAL(im1.n_cols, im2.n_cols);
   BOOST_REQUIRE_EQUAL(im1.n_rows, im2.n_rows);
-  for (size_t i = 10; i < im1.n_elem; ++i)
+  for (size_t i = 0; i < im1.n_elem; ++i)
     BOOST_REQUIRE_EQUAL(im1[i], im2[i]);
 }
 
+/**
+ * Test if the image is saved correctly using API for transpose.
+ */
+BOOST_AUTO_TEST_CASE(SaveImageTransposeAPITest)
+{
+  data::ImageInfo* info = new ImageInfo(5, 5, 3, 90);
+
+  arma::Mat<unsigned char> im1;
+  size_t dimension = info->Width() * info->Height() * info->Channels();
+  im1 = arma::randi<arma::Mat<unsigned char>>(dimension, 1);
+  BOOST_REQUIRE(data::Save("APITest.bmp", im1, *info, false, false) == true);
+
+  arma::Mat<unsigned char> im2;
+  BOOST_REQUIRE(data::Load("APITest.bmp", im2, *info, false, false) == true);
+
+  BOOST_REQUIRE_EQUAL(im1.n_cols, im2.n_cols);
+  BOOST_REQUIRE_EQUAL(im1.n_rows, im2.n_rows);
+  for (size_t i = 0; i < im1.n_elem; ++i)
+    BOOST_REQUIRE_EQUAL(im1[i], im2[i]);
+}
+
+/**
+ * Test that the image is loaded correctly into the matrix using the API
+ * for vectors.
+ */
+BOOST_AUTO_TEST_CASE(LoadVectorImageAPITest)
+{
+  arma::Mat<unsigned char> matrix;
+  data::ImageInfo info;
+  std::vector<std::string> file = {"test_image.png", "test_image.png"};
+  BOOST_REQUIRE(data::Load(file, matrix, info, false,
+      true) == true);
+  BOOST_REQUIRE_EQUAL(matrix.n_rows, 50 * 50 * 3); // width * height * channels.
+  BOOST_REQUIRE_EQUAL(matrix.n_cols, 2);
+}
+
+/**
+ * Test if the image is saved correctly using vector saving API for transpose.
+ */
+BOOST_AUTO_TEST_CASE(SaveImageVectorAPITest)
+{
+  data::ImageInfo info(5, 5, 3);
+
+  arma::Mat<unsigned char> im1;
+  size_t dimension = info.Width() * info.Height() * info.Channels();
+  im1 = arma::randi<arma::Mat<unsigned char>>(dimension, 2);
+  std::vector<std::string> file = {"APITest1.bmp", "APITest2.bmp"};
+  BOOST_REQUIRE(data::Save(file, im1, info, false, false) == true);
+
+  arma::Mat<unsigned char> im2;
+  BOOST_REQUIRE(data::Load(file, im2, info, false, false) == true);
+
+  BOOST_REQUIRE_EQUAL(im1.n_cols, im2.n_cols);
+  BOOST_REQUIRE_EQUAL(im1.n_rows, im2.n_rows);
+  for (size_t i = 0; i < im1.n_elem; ++i)
+    BOOST_REQUIRE_EQUAL(im1[i], im2[i]);
+}
+
+/**
+ * Test if the image is saved correctly using API for arma mat.
+ */
+BOOST_AUTO_TEST_CASE(SaveImageMatAPITest)
+{
+  data::ImageInfo* info = new ImageInfo(5, 5, 3);
+
+  arma::Mat<unsigned char> im1;
+  size_t dimension = info->Width() * info->Height() * info->Channels();
+  im1 = arma::randi<arma::Mat<unsigned char>>(dimension, 1);
+  arma::mat input = arma::conv_to<arma::mat>::from(im1);
+  BOOST_REQUIRE(Save("APITest.bmp", input, *info, false, false) == true);
```

Comment:
I'd consider using `BOOST_REQUIRE_EQUAL()` here, but I guess it doesn't make a huge difference either way.